VideoLocker/src/main/java/org/edx/mobile/player/Player.java

```diff
@@ -552,179 +551,185 @@ public void reset() {
-
-            return lastCurrentPosition;
+        } catch (Exception ex) {
+            logger.error(ex);
+            ;
         }
 
-        @Override
-        public int getDuration() {
-            try {
-                if (state == PlayerState.PREPARED
-                        || state == PlayerState.PLAYING
-                        || state == PlayerState.PAUSED
-                        || state == PlayerState.STOPPED
-                        || state == PlayerState.PLAYBACK_COMPLETE
-                        || state == PlayerState.LAGGING) {
-                    lastDuration = super.getDuration();
-                }
-            } catch(Exception ex) {
-                logger.error(ex);
-            }
-            return lastDuration;
-        }
+        return lastCurrentPosition;
+    }
 
-        @Override
-        public boolean isPlaying() {
-            if (state == PlayerState.PLAYING
+    @Override
+    public int getDuration() {
+        try {
+            if (state == PlayerState.PREPARED
+                    || state == PlayerState.PLAYING
+                    || state == PlayerState.PAUSED
+                    || state == PlayerState.STOPPED
+                    || state == PlayerState.PLAYBACK_COMPLETE
                     || state == PlayerState.LAGGING) {
-                logger.debug("isPlaying = TRUE");
-                return super.isPlaying();
+                lastDuration = super.getDuration();
             }
-            logger.debug("isPlaying = FALSE; state=" + state);
-            return false;
+        } catch (Exception ex) {
+            logger.error(ex);
         }
+        return lastDuration;
+    }
 
-        @Override
-        public boolean isPaused() {
-            if (state == PlayerState.PAUSED) {
-                logger.debug("isPaused = TRUE");
-                return true;
-            }
-            logger.debug("isPaused = FALSE; state=" + state);
-            return false;
-        }
+    @Override
+    public boolean isPlaying() {
+        if (state == PlayerState.PLAYING
+                || state == PlayerState.LAGGING) {
+            logger.debug("isPlaying = TRUE");
+            return super.isPlaying();
+        }
+        logger.debug("isPlaying = FALSE; state=" + state);
+        return false;
+    }
 
-        @Override
-        public void setLMSUrl(String url) {
-            this.lmsURL = url;
+    @Override
+    public boolean isPaused() {
+        if (state == PlayerState.PAUSED) {
+            logger.debug("isPaused = TRUE");
+            return true;
         }
+        logger.debug("isPaused = FALSE; state=" + state);
+        return false;
+    }
 
-        @Override
-        public void setNextPreviousListener(OnClickListener next,
-                OnClickListener prev) {
-            if(controller!=null){
-                controller.setPrevNextListeners(next, prev);
-            }
-        }
+    @Override
+    public boolean isFrozen() {
+        return isFreeze;
```

Comment:
Maybe have the variable renamed to `isFrozen` as well? It's more grammatically correct anyway :wink: